database/postgresql/src/main/java/com/example/javaopencommerce/catalog/PsqlCategoryRepositoryImpl.java

```diff
@@ -2,10 +2,12 @@
 
 import com.example.javaopencommerce.exception.EntityNotFoundException;
 
+import javax.enterprise.context.ApplicationScoped;
 import javax.persistence.EntityManager;
 import javax.persistence.NoResultException;
 
 
+@ApplicationScoped
```

Comment:
Not a fan of that annotation. It would be better to create bean with a @Producer.